mmdet/datasets/pipelines/rotate.py

```diff
@@ -0,0 +1,251 @@
+import cv2
+import mmcv
+import numpy as np
+
+from mmdet.core.mask import BitmapMasks, PolygonMasks
+from mmdet.datasets import PIPELINES
+
+_MAX_LEVEL = 10
+
+
+def level_to_value(level, max_value):
+    """Mapping level to value based on _MAX_LEVEL and max_value."""
+    return (level / _MAX_LEVEL) * max_value
+
+
+def random_negative(value, random_negative_prob):
+    """Randomly negate value based on random_negative_prob."""
+    return -value if np.random.rand() < random_negative_prob else value
+
+
+def bbox2fields():
+    """The key correspondence from bboxes to labels, masks and
+    segmentations."""
+    bbox2label = {
+        'gt_bboxes': 'gt_labels',
+        'gt_bboxes_ignore': 'gt_labels_ignore'
+    }
+    bbox2mask = {
+        'gt_bboxes': 'gt_masks',
+        'gt_bboxes_ignore': 'gt_masks_ignore'
+    }
+    bbox2seg = {
+        'gt_bboxes': 'gt_semantic_seg',
+    }
+    return bbox2label, bbox2mask, bbox2seg
```

Comment:
The above code seems to duplicate with `translate`. Can we remove this duplication?

Reply:
Hi @yhcao6 , yeah it's duplicated here. I thought this might be solved along with the final merged version (e.g. all the aug operations are ready to merge)?

Reply:
ok